src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -763,32 +807,32 @@ void GoogleCalendarSyncAdaptor::updateLocalCalendarNotebooks(int accountId, cons
             QString currDeviceCalendarId = notebook->pluginName().mid(7);
             if (m_serverCalendarIdToCalendarInfo[accountId].contains(currDeviceCalendarId)) {
                 // the server-side calendar exists on the device.
-                // we don't need to purge it, but we may need to update its summary/color details.
-                deviceCalendarIds.append(currDeviceCalendarId);
-                if (notebook->name() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).summary
-                        || notebook->color() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).color
-                        || notebook->description() != m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).description
-                        || notebook->isReadOnly()) {
-                    // calendar information changed server-side.
-                    notebook->setIsReadOnly(false);
-                    notebook->setName(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).summary);
-                    notebook->setColor(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).color);
-                    notebook->setDescription(m_serverCalendarIdToCalendarInfo[accountId].value(currDeviceCalendarId).description);
-                    m_storage->updateNotebook(notebook);
-                    m_storageNeedsSave = true;
+                if (needCleanSync) {
+                    // we are performing a clean sync cycle.
+                    // we will eventually delete and then insert this notebook.
+                    SOCIALD_LOG_DEBUG("queuing clean sync of local calendar" << notebook->name() << currDeviceCalendarId << "for Google account:" << accountId);
+                    m_serverCalendarIdToCalendarInfo[accountId][currDeviceCalendarId].change = GoogleCalendarSyncAdaptor::CleanSync;
```

Comment:
Yes, local modifications will be dropped due to the clean-sync semantics, currently.  Is this a bad idea?  I don't really know how to recover properly from error state after database writes have started, perhaps I'm overlooking something.

For the second thing: the problem is that notebook modifications are immediately flushed to storage (and if we pass the MemoryOnly bool flag, then bad things happen - ghost events everywhere.  I haven't investigated why, I just assume that it's something mkcal specific).  So, if we apply the notebook modifications immediately now, but then the very next network request fails due to connection error, then it's possible that the notebooks will all have been deleted and re-created (due to clean-sync semantics), and have no content / events in them.  Ie, JB#26988